### DISCUSSION TITLE
The same variable name in Cookie and Path used in `route` and `depends` causes conflict

### DISCUSSION THREAD

**Discussion by poul1x** [fastapi/fastapi#8776] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional
from fastapi import FastAPI
from fastapi import Cookie, Path
from fastapi.param_functions import Depends

app = FastAPI()

async def current_user(
    user_id: Optional[str] = Cookie(
        None, alias="USER_ID", min_length=1, max_length=256
    ),
    cookie: Optional[str] = Cookie(
        None, alias="SESSION_ID", min_length=1, max_length=256
    ),
):
    return {
        "user_id": user_id,
        "session_id": cookie,
        "user_name": "Ivan",
    }

@app.get("/{user_id}")
async def some_job(
    current_user: dict = Depends(current_user),
    user_id: str = Path(...),
):
    return {
        "user_id": user_id,
        "user_id_from_cookie": current_user['user_id'],
    }
```

### Description

* Run code, and see at generated documentation. Note: no `USER_ID` in cookies

![image](https://user-images.githubusercontent.com/31695750/128096590-52246ca7-b523-4796-bbfb-35ba0a37890a.png)

* Try to pass `USER_ID`, `SESSION_ID` and `user_id` to test function `some_job`. You'll see that cookie `user_id`  is overriden by path `user_id`.

* Try to change cookie variable name to `_user_id`. This resolves name conflict and generates correct documentation

![image](https://user-images.githubusercontent.com/31695750/128096895-95adeb2d-1b22-465a-8c6c-2dd59e2fb2b0.png)

### Operating System

Linux

### FastAPI Version

0.63.0

### Python Version

Python 3.7.3

**Comment by Dustyposa** [comment]
I don't know if that's a bug.
But I think it's not a good way to do this:
```python
@app.get("/{user_id}")
async def some_job(
    user_id: str = Path(...),
    user_id: Optional[str] = Cookie(
        None, alias="USER_ID", min_length=1, max_length=256
    ),
    cookie: Optional[str] = Cookie(
        None, alias="SESSION_ID", min_length=1, max_length=256
    )
):
```

**Comment by poul1x** [comment]
I didn't noticed, that there was a name conflict.
I think this should be treated as usage error.  
Is it possible to put an assert to avoid such name conflicts?

**Comment by Dustyposa** [comment]
Yes, may be do an enhance to do this.  
The name conflict in two function, may be forget it. 
I will have a try, thanks for your advice.

**Comment by poul1x** [comment]
Thank you
